### PR TITLE
Instrumentation: Make backend plugin metrics endpoints available with optional authentication

### LINKED ISSUE #45935: Backend plugin metrics are behind an authenticated endpoint
**What happened**:

It appears to be impossible to scrape metrics for backend plugins without authenticating. This makes it quite difficult to monitor plugins. Grafana's own `/metrics` endpoint is unauthenticated, which is helpful, but plugin endpoints (like `/api/plugins/grafana-splunk-datasource/metrics`, for example) are not possible to be scraped without authentication.

**What you expected to happen**:

Plugin metrics endpoints should be able to be scraped without authenticating

**How to reproduce it (as minimally and precisely as possible)**:

Run Grafana with a backend plugin running, and `curl` the plugin's metrics endpoint. For example:

```console
$ curl -v http://127.0.0.1:3000/api/plugins/grafana-splunk-datasource/metrics
[...]
> GET /api/plugins/grafana-splunk-datasource/metrics HTTP/1.1
[...]
< HTTP/1.1 401 Unauthorized
[...]
{"message":"Unauthorized"}
```

**Anything else we need to know?**:

**Environment**:
- Grafana version: 8.4.1
- Data source type & version: Any backend datasource
- OS Grafana is installed on: Linux
- User OS & Browser: any
- Grafana plugins:
- Others:

### PR DESCRIPTION
**Pull request by marefr**
**What this PR does / why we need it**:
Make backend plugin metrics endpoints available with optional authentication. Reuses existing [metrics configuration](https://grafana.com/docs/grafana/latest/administration/configuration/#metrics) for enable/disable metrics endpoint and basic auth username/password.

Plugin metrics endpoints available via `/metrics/plugins/<plugin id>`, e.g. http://localhost:3000/metrics/plugins/grafana-test-datasource

**Which issue(s) this PR fixes**:
Fixes #45935

**Special notes for your reviewer**:
- Prometheus default metrics endpoint configuration is `/metrics`. To not have to re-configure this in scrape configuration we might want to change this to `/plugins/<plugin id>/metrics`.
- Would be quite nice if Grafana exposed an [HTTP SD endpoint](https://prometheus.io/docs/prometheus/latest/http_sd/) returning a scrape config for all available backend plugins. Maybe something for later.

### CODE REVIEW

**Comment by hairyhenderson** [comment]
> * Prometheus default metrics endpoint configuration is `/metrics`. To not have to re-configure this in scrape configuration we might want to change this to `/plugins/<plugin id>/metrics`.

I'm not sure that `/plugins/<plugin id>/metrics` would have any advantage over `/metrics/plugins/<plugin id>` - both need a specialized scrape config... There is an advantage to `/metrics/plugins/<plugin id>` though, as it makes controlling access to metrics much simpler (i.e. if you want to block metrics access from outside you just block `/metrics[/*]`).

**Review by hairyhenderson — APPROVED** [review]
LGTM! Thanks @marefr 🙇‍♂️

File: docs/sources/administration/view-server/internal-metrics.md [hunk 1]
```diff
@@ -92,3 +92,48 @@ These instructions assume you have already added Graphite as a data source in Gr
    ```
 
 1. Restart Grafana. Grafana now exposes metrics at http://localhost:3000/metrics and sends them to the Graphite location you specified.
+
+## Pull metrics from Grafana backend plugin into Prometheus
+
+Any installed [backend plugin]({{< relref "../../developers/plugins/backend/_index.md" >}}) exposes a metrics endpoint thru Grafana that you can configure Prometheus to scrape.
```
Comment (hairyhenderson) [inline docs/sources/administration/view-server/internal-metrics.md:98]:
```suggestion
Any installed [backend plugin]({{< relref "../../developers/plugins/backend/_index.md" >}}) exposes a metrics endpoint through Grafana that you can configure Prometheus to scrape.
```

**Review by wbrowne — APPROVED** [review]

File: docs/sources/administration/view-server/internal-metrics.md [hunk 2]
```diff
@@ -92,3 +92,48 @@ These instructions assume you have already added Graphite as a data source in Gr
...
+      metrics_path: /metrics/plugins/grafana-test-datasource
+
+      static_configs:
+        - targets: ['localhost:3000']
+   ```
+
+1. Restart Prometheus. Your new job should appear on the Targets tab.
+1. In Grafana, hover your mouse over the **Configuration** (gear) icon on the left sidebar and then click **Data Sources**.
+1. Select the **Prometheus** data source.
+1. On the Dashboards tab, **Import** a Go processes dashboard.
```
Comment (marefr) [inline docs/sources/administration/view-server/internal-metrics.md:139]: This last line still might feel a bit weird "Go processes dashboard". There are some available at grafana.com/grafana/dashboards. @wbrowne @hairyhenderson WDYT is it okay?

Comment (wbrowne) [reply docs/sources/administration/view-server/internal-metrics.md:139]: Maybe something like?
>"**Import** a Golang application metrics dashboard - for example [Go Processes](https://grafana.com/grafana/dashboards/6671)"

**Pull request merged**